src/components/AnimatedSignal/hooks/useAnimatedSignal.ts

```diff
@@ -0,0 +1,256 @@
+import {
+  interpolate,
+  SharedValue,
+  useAnimatedProps,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+import { makeArcPath } from '../commonUtils';
+
+const useAnimatedSignal = (size: number, numberOfArcs: number) => {
+  const squareSize: number = size / 5;
+  const sizeVal: SharedValue<number> = useSharedValue(0);
+  const arcOneX: SharedValue<number> = useSharedValue(150);
+  const arcOneY: SharedValue<number> = useSharedValue(110);
+  const arcOneStartAngle: SharedValue<number> = useSharedValue(-0.7);
+  const arcOneEndAngle: SharedValue<number> = useSharedValue(-0.7);
+  const arcOneRadius: SharedValue<number> = useSharedValue(40);
+  const arcOneWidth: SharedValue<number> = useSharedValue(5);
+  const arcTwoX: SharedValue<number> = useSharedValue(125);
+  const arcTwoY: SharedValue<number> = useSharedValue(130);
+  const arcTwoStartAngle: SharedValue<number> = useSharedValue(-0.5);
+  const arcTwoEndAngle: SharedValue<number> = useSharedValue(-0.5);
+  const arcTwoRadius: SharedValue<number> = useSharedValue(60);
+  const arcTwoWidth: SharedValue<number> = useSharedValue(5);
+  const arcThreeX: SharedValue<number> = useSharedValue(110);
+  const arcThreeY: SharedValue<number> = useSharedValue(155);
+  const arcThreeStartAngle: SharedValue<number> = useSharedValue(-0.2);
+  const arcThreeEndAngle: SharedValue<number> = useSharedValue(-0.2);
+  const arcThreeRadius: SharedValue<number> = useSharedValue(50);
+  const arcThreeWidth: SharedValue<number> = useSharedValue(5);
+  const arcFourX: SharedValue<number> = useSharedValue(0.9);
+  const arcFourY: SharedValue<number> = useSharedValue(0.9);
+  const arcFourStartAngle: SharedValue<number> = useSharedValue(5.475);
+  const arcFourEndAngle: SharedValue<number> = useSharedValue(7.1);
+  const arcFourRadius: SharedValue<number> = useSharedValue(55);
+  const arcFourWidth: SharedValue<number> = useSharedValue(5);
+  const displayArcTwo: boolean = numberOfArcs > 1;
+  const displayArcThree: boolean = numberOfArcs > 2;
+  const displayArcFour: boolean = numberOfArcs > 3;
+
+  sizeVal.value = withTiming(29, {
+    duration: 1500,
+  });
+
+  const arcOneAnimatedProps = useAnimatedProps(() => {
+    arcOneX.value = interpolate(
+      sizeVal.value,
+      [0, 4, 6.5, 12, 21, 29],
+      [150, 150, 120, 80, 95, 95].map((item) => (item * size) / 300)
+    );
+
+    arcOneY.value = interpolate(
+      sizeVal.value,
+      [0, 4, 6.5, 12, 21, 29],
+      [110, 115, 125, 110, 70, 70].map((item) => (item * size) / 300)
+    );
+
+    arcOneStartAngle.value = interpolate(
+      sizeVal.value,
+      [0, 4, 6.5, 12, 21, 29],
+      [-0.7, -0.7, 1.3, 3.3, 5.475, 5.475]
+    );
+
+    arcOneEndAngle.value = interpolate(
+      sizeVal.value,
+      [0, 4, 6.5, 12, 21, 29],
+      [-0.7, 1.3, 3.3, 5.3, 7.07, 7.07]
+    );
+
+    arcOneRadius.value = interpolate(
+      sizeVal.value,
+      [0, 4, 6.5, 12, 21, 29],
```

Comment:
[0, 4, 6.5, 12, 21, 29] -> This array is used at multiple places with same so you can create an object and use that.

Reply:
Resolved